packages/expo-firebase-analytics/android/src/main/java/expo/modules/firebase/analytics/FirebaseAnalyticsModule.java

```diff
@@ -0,0 +1,176 @@
+package expo.modules.firebase.analytics;
+
+import android.app.Activity;
+import android.content.Context;
+
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.analytics.FirebaseAnalytics;
+
+import expo.modules.firebase.core.FirebaseCoreInterface;
+
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.Promise;
+import org.unimodules.core.arguments.MapArguments;
+import org.unimodules.core.interfaces.ActivityProvider;
+import org.unimodules.core.interfaces.ExpoMethod;
+import org.unimodules.core.interfaces.RegistryLifecycleListener;
+
+import java.util.Map;
+import java.util.Set;
+
+import androidx.annotation.Nullable;
+
+public class FirebaseAnalyticsModule extends ExportedModule implements RegistryLifecycleListener {
+  private static final String NAME = "ExpoFirebaseAnalytics";
+
+  private Activity mActivity;
+  private ModuleRegistry mModuleRegistry;
+
+  public FirebaseAnalyticsModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public void onCreate(ModuleRegistry moduleRegistry) {
+    ActivityProvider mActivityProvider = moduleRegistry.getModule(ActivityProvider.class);
+    mActivity = mActivityProvider.getCurrentActivity();
+    mModuleRegistry = moduleRegistry;
+
+  }
+
+  private FirebaseAnalytics getFirebaseAnalyticsOrReject(final Promise promise) {
+    FirebaseCoreInterface firebaseCore = mModuleRegistry.getModule(FirebaseCoreInterface.class);
+    if (firebaseCore == null) {
+      promise.reject("ERR_FIREBASE_ANALYTICS",
+              "FirebaseCore could not be found. Ensure that your app has correctly linked 'expo-firebase-core' and your project has react-native-unimodules installed.");
+      return null;
+    }
+    FirebaseApp defaultApp = firebaseCore.getDefaultApp();
+    if (defaultApp == null) {
+      promise.reject("ERR_FIREBASE_ANALYTICS",
+              "Firebase app is not initialized. Ensure your app has a valid google-services.json bundled.");
+      return null;
+    }
+    FirebaseApp systemApp = null;
+    try {
+      systemApp = FirebaseApp.getInstance();
+    } catch (Exception e) {
+      // nop
```

Comment:
maybe we should log this, it looks like it will throw an `IllegalStateException` if the app isn't initialized yet

Reply:
This is just for checking whether the default firebase app is initialized. Unfortunately that's not possible without an exception handler here. If no default app exists, then that's not an error situation per-se.